configuration/NI.Analysis.targets

```diff
@@ -0,0 +1,107 @@
+    <RoslynAnalyzer Include="Microsoft.CodeQuality" />
+    <RoslynAnalyzer Include="Microsoft.NetCore" />
+    <RoslynAnalyzer Include="Microsoft.NetFramework" />
+    <RoslynAnalyzer Include="Text" />
+
+    <!-- Microsoft's CA analyzers -->
+    <Analyzer Include="@(RoslynAnalyzer->'$(NuGetPackagesDirectory)\%(Identity).analyzers\2.6.3\analyzers\dotnet\cs\%(Identity).Analyzers.dll')"/>
+    <Analyzer Include="@(RoslynAnalyzer->'$(NuGetPackagesDirectory)\%(Identity).analyzers\2.6.3\analyzers\dotnet\cs\%(Identity).CSharp.Analyzers.dll')"/>
+    <Analyzer Include="$(PkgMicrosoft_VisualStudio_Threading_Analyzers)\analyzers\cs\Microsoft.VisualStudio.Threading.Analyzers.dll"/>
+    <Analyzer Include="$(PkgMicrosoft_CodeQuality_Analyzers)\analyzers\dotnet\cs\Humanizer.dll" />
+
+    <!-- NI's CA analyzers -->
+    <Analyzer Include="$(NuGetPackagesDirectory)\ni.codeanalysis.analyzers\1.0.2\lib\netstandard1.3\NI.CodeAnalysis.Analyzers.dll"/>
+    <Analyzer Include="$(NuGetPackagesDirectory)\ni.codeanalysis.analyzers\1.0.2\lib\netstandard1.3\Utilities.dll"/>
+    <Analyzer Include="$(NuGetPackagesDirectory)\ni.codequality.analyzers\1.0.2\lib\netstandard1.3\NI.CodeQuality.Analyzers.dll"/>
+    <Analyzer Include="$(NuGetPackagesDirectory)\ni.codequality.analyzers\1.0.2\lib\netstandard1.3\Utilities.dll"/>
+    <Analyzer Include="$(NuGetPackagesDirectory)\nationalinstruments.tools.analyzers.naming\1.0.2\lib\netstandard1.3\NationalInstruments.Tools.Analyzers.Naming.dll"/>
+    <Analyzer Include="$(NuGetPackagesDirectory)\nationalinstruments.tools.analyzers.naming\1.0.2\lib\netstandard1.3\NationalInstruments.Tools.Analyzers.Utilities.dll"/>
+
+    <!-- Configuration files for our analyzers -->
+    <!-- By adding the Link attribute, these files will not be shown in the Solution Explorer for .NET Core projects. -->
+    <AdditionalFiles Include="$(NI1704_AdditionalSpellingDictionary)" Condition="Exists('$(NI1704_AdditionalSpellingDictionary)')">
+      <Link>$(CodeAnalysisRuleSet)</Link>
+    </AdditionalFiles>
+    <AdditionalFiles Include="$(CodeAnalysisRuleSet)" Condition="Exists('$(CodeAnalysisRuleSet)')">
+     <Link>$(CodeAnalysisRuleSet)</Link>
+    </AdditionalFiles>
+
+    <AdditionalFiles Include="$(NI1704_GlobalExceptions)" Condition="Exists('$(NI1704_GlobalExceptions)')" />
+  </ItemGroup>
+
+  <!-- By default, the NI.ruleset reports violations as warnings instead of errors. 
+       Set TreatWarningsAsErrors to true in release to generate errors. -->
+  <PropertyGroup Condition="'$(Configuration)' == 'Release'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
+  </PropertyGroup>
+
+  <!-- Disable portions of Roslyn analyzers that would otherwise generate errors in our build. There should be a goal
+       around removing as many of these as possible, but we want to report warnings and elevate warnings as errors in
+       our release builds rather than erroring all the time on everything. -->
+  <ItemGroup>
```

Comment:
This is the content that was in NI.RoslynTransition.targets file that I'm just putting directly in here. I don't see a reason to maintain a separate file, but we could if people feel strongly about it.

Reply:
I think I could go either way here. It doesn't quite seem appropriate in this file, but its so little, it doesn't seem worth worrying about adding here. I think if we expect this to grow significantly, it would be nice to have as a separate file.

Reply:
Yeah I agree. This is the largest it's every going to be, and it may shrink as time goes on, though I'm not sure by how much